### PR TITLE
[stix_cyber_observable] Fix KeyError in parsing Email-Message Observable

### PR DESCRIPTION
**Pull request by kohsawa**
### Proposed changes

* Fix KeyError occurs when creating the observable of the Email-Message with the date field from an STIX Bundle Object which includes it.

### Related issues

* No

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by nor3th** [comment]
In which line does the error occur? Since "date" is not a valid EmailMessage field (https://github.com/OpenCTI-Platform/opencti/blob/master/opencti-platform/opencti-graphql/config/schema/opencti.graphql#L5935-L5941), so my guess is that `date` is throwing the error.

**Comment by kohsawa** [comment]
As I understand it, the code block around where I edited is converting the mapping of the EmailMessage Observable property from the official STIX one to OpenCTI's GraphQL schema.

And as shown below, the STIX official EmailMessage mapping uses `date` instead of `attribute_date`.
> https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_grboc7sq5514

Based on the above, looking at the code before the edit, it's accessing `observable_data["attribute_date"]` when `observable_data`  has the `date` property, which causes a KeyError.

File: pycti/entities/opencti_stix_cyber_observable.py [hunk 1]
```diff
@@ -748,7 +748,7 @@ def create(self, **kwargs):
                     "is_multipart": observable_data["is_multipart"]
                     if "is_multipart" in observable_data
                     else None,
-                    "attribute_date": observable_data["attribute_date"]
+                    "attribute_date": observable_data["date"]
                     if "date" in observable_data
```
Comment (nor3th) [inline pycti/entities/opencti_stix_cyber_observable.py:752]:
```suggestion
                    if "attribute_date" in observable_data
```

File: pycti/entities/opencti_stix_cyber_observable.py [hunk 2]
```diff
@@ -748,7 +748,7 @@ def create(self, **kwargs):
                     "is_multipart": observable_data["is_multipart"]
                     if "is_multipart" in observable_data
                     else None,
-                    "attribute_date": observable_data["attribute_date"]
+                    "attribute_date": observable_data["date"]
```
Comment (nor3th) [inline pycti/entities/opencti_stix_cyber_observable.py:751]:
```suggestion
                    "attribute_date": observable_data["attribute_date"]
```

**Comment by nor3th** [comment]
Hey

Yes, but for pycti it matters how the attributes are called in the opencti platform and in this case, the attribute is called `attribute_date` (https://github.com/OpenCTI-Platform/opencti/blob/master/opencti-platform/opencti-graphql/config/schema/opencti.graphql#L5951-L5957) I guess this alternative is chosen as `date` is already used for the time series. https://github.com/OpenCTI-Platform/opencti/blob/master/opencti-platform/opencti-graphql/config/schema/opencti.graphql#L226

I'll just modify the file directly, but thank you very much for pointing this issue out.

Regards

**Pull request merged**